plugins/json_rpc/plugin.cpp

```diff
@@ -161,92 +161,114 @@ namespace golos {
+                    if (v.size() < 2 || v.size() > 3) {
+                        func_args.error(JSON_RPC_INVALID_REQUEST, "A member \"params\" should be [\"api\", \"method\", \"args\"]");
+                        return nullptr;
+                    }
 
-                        ret = find_api_method(v[0].as_string(), v[1].as_string());
-                        func_args.plugin = v[0].as_string();
-                        func_args.method = v[1].as_string();
-                        fc::variant tmp = (v.size() == 3) ? v[2] : fc::json::from_string("{}");
-                        func_args.args = tmp.as<std::vector<fc::variant>>();
-                    } else {
-                        vector<std::string> v;
-                        boost::split(v, method, boost::is_any_of("."));
+                    if (nullptr == (ret = find_api_method(v[0].as_string(), v[1].as_string(), func_args))) {
+                        return nullptr;
+                    }
 
-                        FC_ASSERT(v.size() == 2, "method specification invalid. Should be api.method");
+                    func_args.plugin = v[0].as_string();
+                    func_args.method = v[1].as_string();
+                    fc::variant tmp = (v.size() == 3) ? v[2] : fc::json::from_string("[]");
 
-                        ret = find_api_method(v[0], v[1]);
-                        func_args.plugin = v[0];
-                        func_args.method = v[1];
-                        fc::variant tmp = (v.size() == 3) ? v[2] : fc::json::from_string("{}");
+                    try {
                         func_args.args = tmp.as<std::vector<fc::variant>>();
+                    } catch (const fc::bad_cast_exception& e) {
+                        func_args.error(JSON_RPC_INVALID_REQUEST, "A member \"args\" should be array", static_cast<const fc::exception&>(e));
+                        return nullptr;
                     }
 
                     return ret;
                 }
 
-                void rpc_jsonrpc(const fc::variant_object &request, msg_pack &msg) {
-                    // TODO: id is optional value or not?
-                    if (request.contains("id")) {
-                        msg.rpc_id(request["id"]);
-                    }
+                void rpc_jsonrpc(const fc::variant &data, msg_pack &msg) {
+                    fc::variant_object request;
 
-                    if (!request.contains("jsonrpc") || request["jsonrpc"].as_string() != "2.0") {
-                        return msg.error(JSON_RPC_INVALID_REQUEST, "jsonrpc value is not \"2.0\"");
-                    } else if (!request.contains("method")) {
-                        return msg.error(JSON_RPC_INVALID_REQUEST, "A member \"method\" does not exist");
-                    }
+                    try {
+                        request = data.get_object();
+
+                        // TODO: id is optional value or not?
+                        if (request.contains("id")) {
```

Comment:
Note: json-rpc spec says it can be optional, but if there is no `id`:
> "The Server MUST NOT reply to a Notification"

(https://www.jsonrpc.org/specification, 4.1)

So if we want to conform spec, we can't respond (even with error) on request with missing `id`.